Update guidelines for aliases considering versions (#157)
grafana/writers-toolkit: docs/sources/writing-guide/front-matter/index.md

> ## Guideline

Reviewer: ```suggestion
## Guidelines
```

Author: Fixed in baf3c0d

> For unversioned projects, to allow content to be easily moved, include an `aliases` entry that refers to the initial published website directory.
> Hugo doesn't create a redirect `.html` file when the directory is already populated with content.
> 
> > **Note:** The published directory is dependent on which `content` subdirectory documentation is synced to in the website repository.
> > For example, documentation synced to a the `content/docs` directory requires the `/docs` prefix.
> 
> For versioned projects, do not include an `aliases` entry that refers to the initial published website directory as the version in the URL path can cause undesirable redirects.
> For example, a redirect from latest content to an old version.

Reviewer: ```suggestion
Unversioned projects:

Include an `aliases` entry that refers to the initial published website directory. Adding an `aliases` entry makes it easier to move content around.
Hugo doesn't create a redirect `.html` file when the directory is already populated with content.

> **Note:** The published directory is dependent on which `content` subdirectory documentation is synced to in the website repository.
> For example, documentation synced to a the `content/docs` directory requires the `/docs` prefix.

Versioned projects:
Do not include an `aliases` entry that refers to the initial published website directory.  The version in the URL path can cause undesirable redirects, such as a redirect from latest content to an old version.
```

Reviewer: Think it's worth adding why it makes it easier to move content around in the unversioned projects section? 
And what about what you should do when you create a new topic and delete the old one - or what happens with aliases when you move content around (i.e. was is their use?)

Author: I've incorporated the suggestion in 8344fd0 and explained the benefit more clearly in 63a221d.

In terms of alias usage, I think that is already explained earlier in the page:
> Technical writers use [Hugo aliases](https://gohugo.io/content-management/urls/#aliases) to create redirects to the current page from other URLs.
> 
> If you specify `aliases` in the frontmatter, Hugo creates a directory that matches the alias entry that contains a single `.html` file.
